wdl-grammar/src/diagnostic.rs

```diff
@@ -83,6 +84,38 @@ pub struct Diagnostic {
     labels: Vec<Label>,
 }
 
+impl Ord for Diagnostic {
```

Comment:
Importantly, this PR includes some unexpected-but-important changes regarding the ordering of diagnostics to make the output of the tests deterministic. This will change the order of many other unrelated tests.